controllers/predicate.go

```diff
@@ -99,3 +100,54 @@ func (p IgnoreSameContentPredicate) handleDeployment(oldObj, newObj *appsv1.Depl
 	delete(oldObj.Annotations, revAnnotation)
 	delete(newObj.Annotations, revAnnotation)
 }
+
+// NodeTaintChangedPredicate filters if node taints have changed
+type NodeTaintChangedPredicate struct {
+	predicate.Funcs
+}
+
+// Update returns true if the node taints have been changed
+func (p NodeTaintChangedPredicate) Update(e event.UpdateEvent) bool {
+	if e.ObjectOld == nil || e.ObjectNew == nil {
+		return false
+	}
+
+	oldNode, ok := e.ObjectOld.(*corev1.Node)
+	if !ok {
+		return false
+	}
+	newNode, ok := e.ObjectNew.(*corev1.Node)
+	if !ok {
+		return false
+	}
+
+	// Compare taints
+	if len(oldNode.Spec.Taints) != len(newNode.Spec.Taints) {
+		return true
+	}
+
+	// Create maps for easier comparison
+	oldTaints := make(map[string]corev1.Taint)
+	newTaints := make(map[string]corev1.Taint)
+
+	for _, taint := range oldNode.Spec.Taints {
+		oldTaints[taint.Key] = taint
+	}
+	for _, taint := range newNode.Spec.Taints {
+		newTaints[taint.Key] = taint
+	}
+
+	// Compare taint maps
+	if len(oldTaints) != len(newTaints) {
+		return true
+	}
```

Comment:
You can probably use `reflect.DeepEquals` or `equality.Semantic.DeepEquals` from `k8s.io/apimachinery/pkg/api/equality `instead of the multi-stage comparison